apps/qubit/modules/user/actions/clipboardSaveAction.class.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class UserClipboardSaveAction extends sfAction
+{
+  public function execute($request)
+  {
+    $allSlugs = $this->context->user->getClipboard()->getAllByClassName();
+
+    if (!count($allSlugs))
+    {
+      $message = $this->context->i18n->__('No items in clipboard to save.');
+      $this->context->user->setFlash('error', $message);
+    }
+    else
+    {
+      // Generate unique password
+      if (false === $password = $this->getUniquePassword())
+      {
+        $message = $this->context->i18n->__('Clipboard ID generation failure. Please try again.');
+        $this->context->user->setFlash('error', $message);
+      }
+
+      $this->saveClipboard($allSlugs, $password);
+
+      // Inform user of progress
+      $messageText = 'Clipboard saved. Clipoard ID is <b>%1%</b>. Please write this number down. ';
```

Comment:
It's a little confusing for me calling it password in one end and ID in the other.

Reply:
Yeah, this is a bit confusing. I called it "password" internally as that's more an accurate description of its purpose (given it's meant to be randomized rather than be a sequential ID).